src/util/ssl_utils.js

```diff
@@ -34,9 +36,17 @@ function verify_ssl_certificate(certificate) {
 
 // Get SSL certificates from local memroy.
 async function get_ssl_certificates() {
-    if (certs === null) {
+    if (certs !== null) {
+        return certs;
+    }
+
+    return certs_read_sem.surround(async () => {
```

Comment:
why do you need the semaphore?

Reply:
I asked it to make sure its safe in case of concurrent calls to `get_ssl_certificates()`